crates/apollo-parser/examples/modify_query_using_parser_and_encoder.rs

```diff
@@ -0,0 +1,122 @@
+use apollo_parser::{
+    ast::{self, AstNode},
+    Parser,
+};
+
+use anyhow::Result;
+
+// This exampel merges the two operation definitions into a single one.
+fn merge_queries() -> Result<apollo_encoder::Document> {
+    let query = r"#
+    query LaunchSite {
+      launches {
+        launches {
+          id
+          site
+        }
+      }
+    }
+
+    query AstronautInfo {
+      user
+      me
+    }
+    #";
+
+    let parser = Parser::new(query);
+    let ast = parser.parse();
+    assert_eq!(ast.errors().len(), 0);
+
+    let doc = ast.document();
+
+    let mut new_query = apollo_encoder::Document::new();
+    let mut sel_set = Vec::new();
+    for def in doc.definitions() {
+        // We want to combine all of our operations into a single one.
+        if let ast::Definition::OperationDefinition(op) = def {
+            let selections: Vec<apollo_encoder::Selection> = op
+                .selection_set()
+                .unwrap()
+                .selections()
+                .map(|sel| sel.try_into())
+                .collect::<Result<Vec<apollo_encoder::Selection>, _>>()?;
+            sel_set.extend(selections)
+        }
+    }
+
+    let op_def = apollo_encoder::OperationDefinition::new(
+        apollo_encoder::OperationType::Query,
+        apollo_encoder::SelectionSet::with_selections(sel_set),
+    );
+    new_query.operation(op_def);
+
+    Ok(new_query)
+}
+
+// This example only includes fields without the `@omitted` directive.
+fn omitted_fields() -> Result<apollo_encoder::Document> {
+    let query = r"#
+    query Products {
+      isbn @omitted
+      title
+      year @omitted
+      metadata @omitted
+      reviews
+      ...details
+    }
+
+    fragment details on ProductDetails {
+        country
+    }
+
+    #";
+
+    let parser = Parser::new(query);
+    let ast = parser.parse();
+    assert_eq!(ast.errors().len(), 0);
+
+    let doc = ast.document();
+
+    let mut new_query = apollo_encoder::Document::new();
+    for def in doc.definitions() {
+        if let ast::Definition::OperationDefinition(op) = def {
+            let mut selection_set = apollo_encoder::SelectionSet::new();
+            for selection in op.selection_set().unwrap().selections() {
+                if let ast::Selection::Field(field) = selection {
+                    if let Some(dir) = field.directives() {
+                        for dir in dir.directives() {
+                            if dir.name().unwrap().source_string() != "omitted" {
+                                selection_set.selection(apollo_encoder::Selection::Field(
+                                    field.clone().try_into()?,
+                                ))
+                            }
+                        }
```

Comment:
this should work I think:
```suggestion
                        let omit = dir.directives().any(|dir| dir.name().unwrap().text() == "omitted");
                        if !omit {
                            selection_set.selection(apollo_encoder::Selection::Field(
                                field.clone().try_into()?,
                            ))
                        }
```

Reply:
ah you're right! thank you so much :)